crates/compiler/mono/src/ir.rs

```diff
@@ -6657,6 +6673,136 @@ fn register_capturing_closure<'a>(
     }
 }
 
+fn get_sorted_tags<'a>(
+    subs: &'a Subs,
+    var: Variable,
+) -> Option<(
+    (impl Iterator<Item = (TagName, &'a [Variable])> + 'a),
+    roc_types::subs::TagExt,
+)> {
+    let content = subs.get_content_without_compacting(var);
+    match content {
+        Content::Structure(FlatType::TagUnion(tags, ext)) => {
+            let (tags, ext) = tags.sorted_iterator_and_ext(&subs, *ext);
+            debug_assert!(ext_var_is_empty_tag_union(subs, ext));
+            Some((tags.map(|(tag_name, variables)| (tag_name, variables)), ext))
+        }
+        Content::Structure(FlatType::RecursiveTagUnion(_var, _tags, _ext)) => {
+            todo!("refine_tag_union: recursive tag union")
+        }
+        _ => None,
+    }
+}
+
+fn unique_symbol(ident_ids: &mut IdentIds, home: ModuleId) -> Symbol {
+    let ident_id = ident_ids.gen_unique();
+
+    Symbol::new(home, ident_id)
+}
```

Comment:
just fyi, there is a `unique_symbol` on the `Env`

Reply:
Yeah, that's what I originally did in `refine_tag_union`, but the borrow checker didn't like that I was both borrowing `env.subs` and mutably borrowing `env`. This `unique_symbol()` acts on `IdentIds` instead of `Env` so that I can  mutably borrow only `env.ident_ids`.